### PR TITLE
Add `pad_to_multiple_of` to `pad_sequence`

### LINKED ISSUE #150989: Add `pad_to_multiple_of` to `torch.nn.utils.rnn.pad_sequence`
### 🚀 The feature, motivation and pitch

Hugging Face tokenizers support [`pad_to_multiple_of`](https://huggingface.co/docs/transformers/en/main_classes/tokenizer#transformers.PreTrainedTokenizerFast.__call__.pad_to_multiple_of), which allows you to pad your sequence's length to a multiple of a number. This comes from a piece of [NVIDIA Tensor Core documentation](https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html#tensor-core-shape):

> ## 4.1. Satisfying Tensor Core Shape Constraints
> Due to their design, Tensor Cores have shape constraints on their inputs. For **matrix multiplication**:
> 
>*  On FP16 inputs, all three dimensions (M, N, K) must be multiples of 8.
> * On INT8 inputs (Turing only), all three dimensions must be multiples of 16.
> 
> For **convolution**:
> 
> * On FP16 inputs, input and output channels must be multiples of 8.
> * On INT8 inputs (Turing only), input and output channels must be multiples of 16.
> 
> In practice, for mixed precision training, our recommendations are:
> 
> 1. Choose mini-batch to be a multiple of 8
> 2. Choose linear layer dimensions to be a multiple of 8
> 3. Choose convolution layer channel counts to be a multiple of 8
> 4. For classification problems, pad vocabulary to be a multiple of 8
> 5. **For sequence problems, pad the sequence length to be a multiple of 8**

It would be convenient for `torch.nn.utils.rnn.pad_sequence` to support `pad_to_multiple_of`.

### Alternatives

My hack right now is to manually create a sequence of my desired length and then discard it from my padded sequence, which is less efficient than the proposed solution.

```python
    if pad_to_multiple_of is not None:
        max_length = max(seq.shape[-1] for seq in sequences)
        max_length = (max_length + pad_to_multiple_of - 1) // pad_to_multiple_of * pad_to_multiple_of
        sequences.append(torch.empty(max_length, dtype=dtype, device=device))
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ringohoffman**
Fixes #150989

`pad_to_multiple_of=8` should be used  to create sequences that take advantage of NVIDIA Tensor Cores when using mixed precision on GPUs with compute capability >= 7.5 (Volta).

### CODE REVIEW

**Comment by ringohoffman** [comment]
Putting this in draft to land after https://github.com/pytorch/pytorch/pull/151021
